### PR TITLE
Simplify

### PR DESCRIPTION
**Pull request by Veetaha**
I absolutely love tha fact that removing `.clone()` simplifies the code comparing to other languages where it's actually the contrary (ahem ~~`std::move()`~~)

### CODE REVIEW

File: crates/ra_hir_ty/src/display.rs [hunk 1]
```diff
@@ -321,8 +319,8 @@ impl HirDisplay for Ty {
     }
 }
 
-fn write_bounds_like_dyn_trait(
-    predicates: &[GenericPredicate],
+fn write_bounds_like_dyn_trait<'a>(
+    predicates: impl IntoIterator<Item = impl Borrow<GenericPredicate>>,
```
Comment (Veetaha) [inline crates/ra_hir_ty/src/display.rs]: That feel, when one invocation passes `Item = GenericPredicate`, but the other is `Item = &GenericPredicate`. Not sure if my solution is overengineered, I can leave intermediate collect::<Vec<_>> if you wish

File: crates/ra_hir_ty/src/display.rs [hunk 2]
```diff
@@ -322,7 +320,7 @@ impl HirDisplay for Ty {
 }
 
 fn write_bounds_like_dyn_trait(
-    predicates: &[GenericPredicate],
+    predicates: impl IntoIterator<Item = impl Borrow<GenericPredicate>>,
```
Comment (Veetaha) [inline crates/ra_hir_ty/src/display.rs]: That feel, when one invocation passes `Item = GenericPredicate`, but the other is `Item = &GenericPredicate`. Not sure if my solution is overengineered, I can leave intermediate `.collect::<Vec<_>>()` if you wish

Comment (matklad) [reply crates/ra_hir_ty/src/display.rs]: I'd rather keep this function non-generic, it's rather large and def not hot, so we should avoid monomorphisation.

Comment (Veetaha) [reply crates/ra_hir_ty/src/display.rs]: I would think it's hot since inlay hints are recomputed on each character typed, but if you insist

Comment (matklad) [reply crates/ra_hir_ty/src/display.rs]: Anything related to presentation is not hot (or can be made not hot), as we present only few things to the user. 

Ie, `O(characters on the scren)` things are cold, `O(code in the project)` things ar hot.

Comment (Veetaha) [reply crates/ra_hir_ty/src/display.rs]: Done

**Pull request merged**